### PR TITLE
feat: Connection pooling CE

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.
>
#### PR fixes following issue(s)
Fixes # (issue number)
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/components/utils/ReduxFormTextField.tsx [hunk 1]
```diff
@@ -20,11 +20,9 @@ const renderComponent = (
       {...omit(componentProps, "type")}
       {...componentProps.input}
       errorMessage={
-        !componentProps.hideErrorMessage &&
-        showError &&
-        componentProps.meta.error &&
-        componentProps.meta.error
+        !componentProps.hideErrorMessage && componentProps.meta.error
```
Comment (ankitakinger) [inline app/client/src/components/utils/ReduxFormTextField.tsx:23]: why exactly are we changing this?

File: app/client/src/ce/pages/AdminSettings/config/DeveloperSettings/index.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import type { AdminConfigType } from "@appsmith/pages/AdminSettings/config/types";
+import {
+  CategoryType,
+  SettingCategories,
+  SettingTypes,
+} from "@appsmith/pages/AdminSettings/config/types";
+import { googleMapsConfig } from "./googleMaps";
```
Comment (ankitakinger) [inline app/client/src/ce/pages/AdminSettings/config/DeveloperSettings/index.ts:7]:
```suggestion
import { googleMapsConfig } from "@appsmith/pages/AdminSettings/config/DeveloperSettings/googleMaps";
```

File: app/client/src/ce/pages/AdminSettings/BreadcrumbCategories.tsx [hunk 3]
```diff
@@ -15,9 +15,11 @@ export const BreadcrumbCategories = {
...
     text: "Email",
   },
-  [SettingCategories.GOOGLE_MAPS]: {
-    href: adminSettingsCategoryUrl({ category: SettingCategories.GOOGLE_MAPS }),
-    text: "Google Maps",
+  [SettingCategories.DEVELOPER_SETTINGS]: {
+    href: adminSettingsCategoryUrl({
+      category: SettingCategories.DEVELOPER_SETTINGS,
+    }),
+    text: "Developer Settings",
```
Comment (ankitakinger) [inline app/client/src/ce/pages/AdminSettings/BreadcrumbCategories.tsx:22]: use sentence casing

File: app/client/src/ce/pages/AdminSettings/config/DeveloperSettings/googleMaps.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { GOOGLE_MAPS_SETUP_DOC } from "constants/ThirdPartyConstants";
+import type { Setting } from "../types";
+import { SettingCategories, SettingSubtype, SettingTypes } from "../types";
```
Comment (ankitakinger) [inline app/client/src/ce/pages/AdminSettings/config/DeveloperSettings/googleMaps.ts:3]:
```suggestion
import { SettingCategories, SettingSubtype, SettingTypes } from "@appsmith/pages/AdminSettings/config/types";
```

File: app/client/src/ce/pages/AdminSettings/config/DeveloperSettings/index.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+export const config: AdminConfigType = {
+  icon: "snippet",
+  type: SettingCategories.DEVELOPER_SETTINGS,
+  categoryType: CategoryType.GENERAL,
+  controlType: SettingTypes.GROUP,
+  title: "Developer Settings",
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/pages/AdminSettings/config/DeveloperSettings/index.ts]: done

File: app/client/src/ce/pages/AdminSettings/BreadcrumbCategories.tsx (hunk 3 above)
Comment (ayushpahwa) [reply app/client/src/ce/pages/AdminSettings/BreadcrumbCategories.tsx]: done

File: app/client/src/components/utils/ReduxFormTextField.tsx (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/components/utils/ReduxFormTextField.tsx:23]: This is just a refactor

**Review by ankitakinger — DISMISSED** [review]
LGTM

**Comment by ayushpahwa** [comment]
Adding skip testplan label since this is just supporting pr for code split. They main EE PR is tested already.

**Review by dipyamanbiswas07 — APPROVED** [review]

**Pull request merged**